core/src/main/java/org/infinispan/functional/impl/MetaParams.java

```diff
@@ -288,6 +290,25 @@ void merge(MetaParams other) {
       return new It();
    }
 
+   @Override
+   public Spliterator<MetaParam<?>> spliterator() {
+      return Spliterators.spliterator(iterator(), length, Spliterator.DISTINCT | Spliterator.NONNULL);
+   }
+
+   public MetaParam.Writable[] toWritableMetas() {
+      int writable = 0;
```

Comment:
Btw., I think that @karesti will introduce Metadata setter to the func API, and this will be another use case for that since the double conversion is unfortunate.